xarray/core/dataset.py

```diff
@@ -1657,6 +1657,45 @@ def broadcast_equals(self, other: Dataset) -> bool:
         the other dataset can still be broadcast equal if the the non-scalar
         variable is a constant.
 
+        Example
+        -------
+
+        # Create two datasets
+        >>> dataset1 = xr.Dataset({"temperature": xr.DataArray([20, 25, 30])})
+        >>> dataset2 = xr.Dataset({"temperature": xr.DataArray([20, 25, 30])})
+
+        # Check if the datasets are equal after broadcasting
+        >>> is_equal = (dataset1 == dataset2).all()
+        >>> is_equal
+        <xarray.Dataset>
```

Comment:
So you're not actually using the `.broadcast_equals` method here.

You want to do something involving `ds1.broadcast_equals(ds2)`